plugins/plugin-gdb/che-plugin-gdb-server/src/main/java/org/eclipse/che/plugin/gdb/server/GdbDebugger.java

```diff
@@ -0,0 +1,383 @@
+            if (path.isEmpty()) {
+                throw new DebuggerException("Variable path is empty");
+            }
+            gdb.setVar(path.get(0), variable.getValue());
+        } catch (IOException | GdbParseException | InterruptedException e) {
+            throw new DebuggerException("Can't set value for " + variable.getName(), e);
+        }
+    }
+
+    @Override
+    public Value getValue(VariablePath variablePath) throws DebuggerException {
+        try {
+            List<String> path = variablePath.getPath();
+            if (path.isEmpty()) {
+                throw new DebuggerException("Variable path is empty");
+            }
+
+            GdbPrint gdbPrint = gdb.print(path.get(0));
+
+            Value value = newDto(Value.class);
+            value.setValue(gdbPrint.getValue());
+            return value;
+        } catch (IOException | GdbParseException | InterruptedException e) {
+            throw new DebuggerException("Can't get value for " + variablePath, e);
+        }
+    }
+
+    @Override
+    public String evaluate(String expression) throws DebuggerException {
+        try {
+            GdbPrint gdbPrint = gdb.print(expression);
+            return gdbPrint.getValue();
+        } catch (IOException | GdbParseException | InterruptedException e) {
+            throw new DebuggerException("Can't evaluate '" + expression + "'", e);
+        }
+    }
+
+    /**
+     * Dump frame.
+     */
+    @Override
+    public StackFrameDump dumpStackFrame() throws DebuggerException {
+        StackFrameDump gdbStackFrameDump = newDto(StackFrameDump.class);
+
+        try {
+            Map<String, String> locals = gdb.infoLocals().getVariables();
+            locals.putAll(gdb.infoArgs().getVariables());
+
+            List<Variable> variables = new ArrayList<>();
+            for (Map.Entry<String, String> e : locals.entrySet()) {
+                String varName = e.getKey();
+                String varValue = e.getValue();
+                String varType;
+                try {
+                    varType = gdb.ptype(varName).getType();
+                } catch (GdbParseException pe) {
+                    LOG.warn(pe.getMessage(), pe);
+                    varType = "";
+                }
+
+                VariablePath variablePath = newDto(VariablePath.class);
+                variablePath.setPath(singletonList(varName));
+
+                Variable variable = newDto(Variable.class);
+                variable.setName(varName);
+                variable.setValue(varValue);
+                variable.setType(varType);
+                variable.setVariablePath(variablePath);
+                variable.setExistInformation(true);
+                variable.setPrimitive(true);
+
+                variables.add(variable);
+            }
+
+            gdbStackFrameDump.setFields(Collections.emptyList());
+            gdbStackFrameDump.setLocalVariables(variables);
```

Comment:
I think you can create `StackFrameDump gdbStackFrameDump = newDto(StackFrameDump.class)` dto and return it at this moment, because it is not necessary to keep the variable as it is not used + i almost sure that you don't need to set `emptyList()` to the fields as DTO never returns null when collection is requested, unless you want to save a small amount of memory.
So the code becomes:
```java
StackFrameDump gdbStackFrameDump = newDto(StackFrameDump.class);
gdbStackFrameDump.setLocalVariables(variables);
return gdbStackFrameDump;
```

It is up to you, but i prefer to add `with*` method and change the code to the following:
```java
return newDto(StackFrameDump.class).withVariables(variables);
```